src/v2/user_account.rs

```diff
@@ -0,0 +1,214 @@
+            path: "/_synapse/admin/v2/users/:user_id",
+            rate_limited: false,
+            authentication: AccessToken,
+        }
+
+        request: {
+            /// user ID
+            #[ruma_api(path)]
+            pub user_id: &'a UserId,
+        }
+
+        #[derive(Default)]
+        response: {
+            /// Information about the user.
+            #[ruma_api(body)]
+            pub info: UserInformation,
+        }
+
+        error: client::Error
+
+    }
+
+    impl<'a> Request<'a> {
+        /// Creates an `Request` with the given user ID.
+        pub fn new(user_id: &'a UserId) -> Self {
+            Self { user_id }
+        }
+    }
+
+    impl Response {
+        /// Creates a new `Response` with all parameters defaulted.
+        pub fn new() -> Self {
+            Default::default()
+        }
+    }
+}
+
+
+pub mod create_modify_user {
+    //! [GET /_synapse/admin/v2/users/:user_id](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst#create-or-modify-account)
+    use super::*;
+
+    ruma_api! {
+        metadata: {
+            description: "create or modify account endpoint",
+            method: PUT,
+            name: "create_modify_account",
+            path: "/_synapse/admin/v2/users/:user_id",
+            rate_limited: false,
+            authentication: AccessToken,
+        }
+
+        request: {
+            /// user ID for the account to renew
+            #[ruma_api(path)]
+            pub user_id: &'a UserId,
+
+            /// This is an optional parameter. Add this parameter to create an account or set this
+            /// password as new one for an existing account.
+            pub password: Option<String>,
+
+            // NOTE: Server explodes if attributes are not omitted but specified as null, like the default
+            // Serde case.
+
+            /// defaults to user_id, or the current value if user already exists
+            /// Some("") is treated as setting it to null.
+            #[serde(skip_serializing_if="Option::is_none")]
+            pub displayname: Option<String>,
+
+            /// defaults to empty, or the current value if user already exists
+            #[serde(skip_serializing_if="Option::is_none")]
+            pub threepids: Option<Vec<super::ThirdPartyIdentifier>>,
+
+            /// The user's avatar URL, if set.
+            #[serde(skip_serializing_if="Option::is_none")]
+            pub avatar_url: Option<String>,
+
+            /// Should the user be a server admin
+            /// defaults to false, or the current value if user already exists
+            #[serde(skip_serializing_if="Option::is_none")]
+            pub admin: Option<bool>,
+
+            /// Should the user be deactivated
+            /// defaults to false, or the current value if user already exists
+            #[serde(skip_serializing_if="Option::is_none")]
+            pub deactivated: Option<bool>,
+
+        }
+
+        #[derive(Default)]
```

Comment:
Since `UserInformation` doesn't implement `Default`, this won't work.

Reply:
I implemented the default traif for this stuct, is this not sufficent?

Reply:
Oh, sorry! I didn't see a derive so thought there wouldn't be an implementation.

Reply:
I just had a look. Your implementation unconditionally recurses into itself, and also given the type's nature I don't think there should be a `Default` impl. Generally that should only exist for types where there is an obvious default; this also goes for the request and response types in `ruma_api!` (so it should be used when there are no parameters or they're all optional / have default values).

Reply:
Ok, but how to implement the Response constructor then?

Reply:
By adding a `UserInformation` parameter to it.